static/js/controllers/inbox.js

```diff
@@ -202,7 +212,11 @@ require('moment/locales');
         });
       };
 
-      $scope.$on('ContactUpdated', function() {
+      $scope.$on('ContactUpdated', function(contact) {
+        if (!contact || contact._deleted) {
+          // allow the ContactCtrl to handle contact deletion
+          return;
+        }
```

Comment:
Happy to roll this back, and add `timeout: false` for the facilities request.  But I can't make the current `develop` branch behave properly for this anyway when a contact is deleted, so it's very hard for me to tell whether things are getting better or not.
N.B. allowing `$scope.updateAvailableFacilities();` does not throw an exception even without `timeout: false`